### ISSUE TITLE
URL scheme 'http' is hardcoded

### ISSUE THREAD

**Issue by Idleworks** [AOEpeople/crawler#92]
In some cases it is necessary to call pages with https instead of http, especially on secured sites where port 80 is blocked.

URL scheme could be fetched from page's url_scheme property, baseUrl or newly created setting in extension manager.

**Comment by tomasnorre** [comment]
Hi @Idleworks 

Thanks for your suggestion it makes perfectly sense to adapt this.

**Comment by ogrosko** [comment]
Hey
Has this feature something in common with my problem:
When I run Crawler queue task from typo3 BE scheduler module my queued url is 
`https://mydomain.com/index.php?id=1` and url is crawled correctly

but wehn I run crawler queue task from cli `./typo3/cli_dispatch.phpsh scheduler -i 3 -f
` (3 is the id of crawler task in scheduler) queued url is `http://./index.php?id=1` and after crawling url I get an error.

Where is the problem. Missed I some configuration for ssl connection?
Thanks for any advice

**Comment by ogrosko** [comment]
Ok I make some investigation:

1. `displayCond` config for `base_url` field in `/typo3conf/ext/crawler/Configuration/TCA/tx_crawler_configuration.php` doesn't work. I have to change it from 
` 'displayCond' => 'FIELD:sys_domain_base_url:=:0',`
to
` 'displayCond' => 'FIELD:sys_domain_base_url:<=:0',`
now it works and when I use this field and set my domain to `https://mydomain.com/`

But it still not working when you use `sys_domain_base_url` (Get Baseurl from Domainrecord) field
It doesn't accept https. And this is object of this task ...

**Comment by tomasnorre** [comment]
Hi @ogrosko 

Thanks for your elaboration we will take a look at it.

**Comment by CDRO** [comment]
Hey people of AOE,

thanks for taking care of the TYPO3 crawler :-)

But it seems that in the latest release, it's impossible again to work with websites using https? Or am I missing something in my config?

Best Regards
Tizian

**Comment by tomasnorre** [comment]
Hi @CDRO 

It should be possible.
Can you please elaborate a little more ?

We are doing it in other project, our configuration is using direct_requests.

**Comment by CDRO** [comment]
My setup is are as follows:
- PHP 7
- TYPO3 7.6
- Linux Ubuntu with Apache 2.4 or Windows with Apache 2.4/IIS 7-8

The extension is configured to not make direct requests since in the past, this led to some major issues  (and also doesn't seem to work on Windows systems) and for the sake of keeping it all homogenous we usually go with the non-direct request approach. Now since we want to force HTTPS wherever we can, we'd also like to be able to do so for the crawler.

I know that in past instances (I'm talking crawler version ≈3) it was possible to set the domain or base url for the configuration by hand (and I'm not longing for these times since it was a mess when you changed the dev domain and had 20+ configurations), this is now way better but I'd really like to have the possibility to choose what protocol the crawler should use when not working in direct_request mode.

I'll write a pull request for this asap, if anybody is interested in it.

Regards
Tizian

**Comment by tomasnorre** [comment]
@CDRO I don't get why this is not working. We are using https for crawler with directRequests.. 
But it's not windows. 

I would have to dig more into this..

**Comment by mgrundkoetter** [comment]
As far as I can see, those lines in the `CrawlerController` are the problem:
```php
$forceSsl = ($pageRow['url_scheme'] === 2) ? true : false;
$res = $this->getUrlsForPageId($pageRow['uid'], $forceSsl);
```
You set `protected function getUrlsForPageId($id, $forceSsl = false)` and it only becomes true if `https` is set on the very page record. So if the whole site uses SSL and it is not set on every single page row, it will be false by default in the crawler. My suggestion is to add a new checkbox to `tx_crawler_configuration` and let the backend admin activate it for the whole config.

**Comment by tomhatzer** [comment]
Hi @tomasnorre !

Do you possibly have any news on this? Is there anything we can help you with?

Thank you very much!

**Comment by tomasnorre** [comment]
@tomhatzer Sorry, didn't have the time for it yet. 
A work around would be to set the "url_scheme" to ssl for every page, direct in the Database. 

The easiest way for us to continue is with a pull request, I have seen the suggestion from @mgrundkoetter, but didn't get to the implementation yet.

**Comment by tomasnorre** [comment]
@tomhatzer Please check https://github.com/AOEpeople/crawler/pull/299 It's not perfekt. But I think it solves your problem.

**Comment by tomasnorre** [comment]
@tomhatzer @Idleworks 
As a fix is merged into master (#299), I will close this issue.

Please reopen it, if it did not solve your problem.
